projects/packages/sync/src/class-dedicated-sender.php

```diff
@@ -70,23 +74,71 @@ public static function spawn_sync( $queue ) {
 			return new WP_Error( 'sync_throttled_' . $queue->id );
 		}
 
+		$url  = rest_url( 'jetpack/v4/sync/spawn-sync' );
+		$url  = add_query_arg( 'time', time(), $url ); // Enforce Cache busting.
 		$args = array(
 			'cookies'   => $_COOKIE,
-			'body'      => array(
-				'jetpack_dedicated_sync_request' => 1,
-			),
 			'blocking'  => false,
 			'timeout'   => 0.01,
 			/** This filter is documented in wp-includes/class-wp-http-streams.php */
 			'sslverify' => apply_filters( 'https_local_ssl_verify', false ),
 		);
 
-		$result = wp_remote_post( site_url(), $args );
+		$result = wp_remote_get( $url, $args );
 		if ( is_wp_error( $result ) ) {
 			return $result;
 		}
 
 		return true;
 	}
 
+	/**
+	 * Test Sync spawning functionality by making a request to the
+	 * Sync spawning endpoint and storing the result (status code) in a transient.
+	 *
+	 * @since $$next_version$$
+	 *
+	 * @return bool True if we got a successful response, false otherwise.
+	 */
+	public static function can_spawn_dedicated_sync_request() {
+		$dedicated_sync_check_transient = self::DEDICATED_SYNC_CHECK_TRANSIENT;
+
+		$dedicated_sync_response_body = get_transient( $dedicated_sync_check_transient );
+
+		if ( false === $dedicated_sync_response_body ) {
+			$url  = rest_url( 'jetpack/v4/sync/spawn-sync' );
+			$url  = add_query_arg( 'time', time(), $url ); // Enforce Cache busting.
+			$args = array(
+				'cookies'   => $_COOKIE,
+				'timeout'   => 30,
+				/** This filter is documented in wp-includes/class-wp-http-streams.php */
+				'sslverify' => apply_filters( 'https_local_ssl_verify', false ),
+			);
+
+			$response                     = wp_remote_get( $url, $args );
+			$dedicated_sync_response_code = wp_remote_retrieve_response_code( $response );
+			$dedicated_sync_response_body = trim( wp_remote_retrieve_body( $response ) );
```

Comment:
@bisko This is guaranteed to always be a string, right? I'm trying to think of any cases that a completely unexpected response could break this.. eg an unreasonably long response body which could cause `set_transient` to fail?

Reply:
Thanks for catching that! Updated!